packages/shared/src/components/squads/SharePostBar.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { FormEvent, ReactElement, useState } from 'react';
+import classNames from 'classnames';
+import { useMutation } from 'react-query';
+import { ProfilePicture } from '../ProfilePicture';
+import { Button, ButtonSize } from '../buttons/Button';
+import { useAuthContext } from '../../contexts/AuthContext';
+import useSidebarRendered from '../../hooks/useSidebarRendered';
+import { getPostByUrl, Post } from '../../graphql/posts';
+
+export interface NewSquadPostProps {
+  post?: Post;
+  url?: string;
+}
+
+interface SharePostBarProps {
+  className?: string;
+  onNewSquadPost?: (props?: NewSquadPostProps) => void;
+}
+
+function SharePostBar({
+  className,
+  onNewSquadPost,
+}: SharePostBarProps): ReactElement {
+  const [url, setUrl] = useState('');
+  const { mutateAsync: getPost } = useMutation(getPostByUrl, {
+    onSuccess: (post) => {
+      onNewSquadPost({ post });
+    },
+    onError: () => {
+      onNewSquadPost({ url });
+    },
```

Comment:
Depending on the error we might want to show the error already here.
So example being invalid URL or deleted post.

Reply:
Right. Let me catch it.